docs(subject builder) : docs for the transport and sasl subject builder (#3037)
kroxylicious/kroxylicious: kroxylicious-docs/docs/_modules/sasl-inspection/con-example-proxy-config.adoc

  - name: my-sasl-inspection-filter
    type: SaslInspection
    config:
      enabledMechanisms: [ "OAUTHBEARER" ]
      requireAuthentication: true
      subjectBuilder: <class name>
      subjectBuilderConfig: <object providing configuration for the subject builder>
----
where:

* `enabledMechanisms` restricts the filter to the given SASL mechanism(s).
Refer to SASL mechanism by its name given in the xref:supported-mechansisms-{context}[supported mechanisms] table.
If omitted, the SASL mechanisms `SCRAM-SHA-256`, `SCRAM-SHA-512` and `OAUTHBEARER` will be enabled by default.

> *  if `requireAuthentication` true then successful authentication will be required before the filter will forward any requests other than those strictly required to perform SASL authentication.

Reviewer: ```suggestion
*  if `requireAuthentication` is true, successful authentication is required before the filter forwards any requests other than those strictly required to perform SASL authentication.
```
